maskrcnn_benchmark/structures/segmentation_mask.py

```diff
@@ -8,68 +11,213 @@
 FLIP_TOP_BOTTOM = 1
 
 
-class Mask(object):
-    """
-    This class is unfinished and not meant for use yet
-    It is supposed to contain the mask for an object as
-    a 2d tensor
-    """
+''' ABSTRACT
+Segmentations come in either:
+1) Binary masks
+2) Polygons
+
+Binary masks can be represented in a contiguous array
+and operations can be carried out more efficiently,
+Therefore BinaryMaskList handles them together.
+
+Polygons are handled separately for each instance,
+by PolygonInstance and instances are handled by
+PolygonList.
+
+SegmentationList is supposed to represent both,
+therefore it wraps the functions of BinaryMaskList
+and PolygonList to make it transparent.
+'''
+
+
+
+class BinaryMaskList(object):
+    '''
+    This class handles binary masks for all objects in the image
+    '''
+
+    def __init__(self, masks, size):
+        '''
+            Arguments:
+                masks: Either torch.tensor of [num_instances, H, W]
+                    or list of torch.tensors of [H, W] or
+                    BinaryMaskList.
+                size: absolute image size, width first
+        '''
+
+        if isinstance(masks, torch.Tensor):
+            pass
+        elif isinstance(masks, (list, tuple)):
+            masks = torch.stack(masks, dim=2)
```

Comment:
I would like to add support for converting RLE masks to binary mask as well.

Reply:
In PR #150 RLE masks are discussed, however I could not find anywhere in the code of RLE support.
In my opinion, the COCO dataset seems to go fine without RLE, and I don't see the point why would it be generally useful.

Aside from that, you can implement the `RLEList` to back up `SegmentationMask` supporting that format, without interfering with the other classes. Of course to support full compatibility, you would have to take care of the to-from conversion for both `RLEList <-> BinaryMaskList` and `RLEList <-> PolygonList`

Reply:
Again `RLE -> polygons` is not always possible considering a donut-like shape. So just convert it into binary mask in the input of binary mask class is enough.

Reply:
Sounds reasonable, I will add it.

Reply:
I have found the implementation for RLE conversions [here](https://github.com/cocodataset/cocoapi/blob/master/PythonAPI/pycocotools/_mask.pyx), which suggested that RLEs are passed as list of dicts, each of them has at least an entry with the key `count`, so I am checking that in the initialization.

@wangg12 could you please run a test if it runs OK this way?